### PR TITLE
Token::getNameForId

### PR DESCRIPTION
**Pull request by SpacePossum**

### CODE REVIEW

File: src/Tokenizer/Token.php [hunk 1]
```diff
@@ -256,11 +256,27 @@ public function getName()
...
+    public static function getNameForId($id)
+    {
+        if (CT::has($id)) {
+            return CT::getName($id);
         }
 
-        return token_name($this->id);
+        $name = token_name($id);
+
+        return 'UNKNOWN' === $name ? null : $name;
```
Comment (SpacePossum) [inline src/Tokenizer/Token.php]: https://github.com/php/php-src/pull/2639

**Comment by SpacePossum** [comment]
closing after discussion

**Comment by keradus** [comment]
are you refering to smalltalk about PR for php-src ?
those PRs are not one-to-one

**Comment by SpacePossum** [comment]
oh I mixed up the PR's ;)

**Comment by keradus** [comment]
cheer up and get well ;)

File: tests/Tokenizer/TokensTest.php [hunk 2]
```diff
@@ -1002,6 +1002,39 @@ public function provideEnsureWhitespaceAtIndexCases()
         ];
     }
 
+    public function testAssertTokensAfterChanging()
```
Comment (julienfalque) [inline tests/Tokenizer/TokensTest.php]: Is this test related to other changes in this PR?

Comment (keradus) [inline tests/Tokenizer/TokensTest.php]: no, and shall go to 2.2 line

File: tests/Tokenizer/TokenTest.php [hunk 3]
```diff
@@ -436,4 +436,29 @@ public function provideIsKeyCaseSensitive()
...
+    {
+        return [
+            [
+                null,
+                -1,
+            ],
+            [
+                'T_CLASS',
+                T_CLASS,
+            ],
```
Comment (keradus) [inline tests/Tokenizer/TokenTest.php]: we shall have CT case as well

Comment (SpacePossum) [inline tests/Tokenizer/TokenTest.php]: :+1:  added

**Comment by SpacePossum** [comment]
lemme know if retarget 2.2 or split up (new test to 2.2 rest to master), all good for me

**Comment by julienfalque** [comment]
I would split up.

**Comment by SpacePossum** [comment]
alrighty, will do :)  :+1:

File: tests/Tokenizer/TokensTest.php (hunk 2 above)
Comment (SpacePossum) [inline tests/Tokenizer/TokensTest.php]: I removed and re-added this. The test covers a part of `AssertTokensTrait` which is not available on 2.2 (introduced on master only)

**Comment by SpacePossum** [comment]
rebased, I think this one is ready.

**Comment by keradus** [comment]
Thank you @SpacePossum.
